lib/index.ts

```diff
@@ -499,6 +507,19 @@ export function emit(rootDecl: TopLevelDeclaration, rootFlags = ContextFlags.Non
                     printObjectTypeMembers(e.members);
                     break;
 
+                case "function":
+                    writeFunctionType(e);

```

Comment:
I think the only correct thing to do here is emit `typeof ${e.name}` since the originating declaration might have overloads (once overloads are supported, of course). It seems like the intent is to allow anonymous function types but those should be their own top-level type kind instead of inlining the signature